core/src/main/java/eu/fasten/core/maven/data/Pom.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2021 Delft University of Technology
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.fasten.core.maven.data;
+
+import static org.apache.commons.lang3.builder.ToStringStyle.MULTI_LINE_STYLE;
+
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import eu.fasten.core.data.Constants;
+import eu.fasten.core.data.collections.ImmutableEmptyLinkedHashSet;
+
+public class Pom {
+
+    private static final LinkedHashSet<Dependency> NO_DEPS = new ImmutableEmptyLinkedHashSet<>();
+    private static final Set<Dependency> NO_DEPMGMT = Set.of();
+
+    public transient long id;
+
+    public final String forge = Constants.mvnForge;
+
+    public final String artifactId;
+    public final String groupId;
+    public final String packagingType;
+    public final String version;
+
+    // g:a:packaging:version
+    public final String parentCoordinate;
+
+    public final long releaseDate;
+    public final String projectName;
+
+    public final Set<Dependency> dependencies;
+    public final Set<Dependency> dependencyManagement;
+
+    public final String repoUrl;
+    public final String commitTag;
+    public final String sourcesUrl;
+    public final String artifactRepository;
+
+    private final int hashCode;
+    private final GAV gav;
+    private final GA ga;
+
+    // use LinkedHashSet for dependencies, because order is relevant for resolution
+    public Pom(String groupId, String artifactId, String packagingType, String version, String parentCoordinate,
+            long releaseDate, String projectName, LinkedHashSet<Dependency> dependencies,
+            Set<Dependency> dependencyManagement, String repoUrl, String commitTag, String sourcesUrl,
+            String artifactRepository) {
+        this.groupId = Ids.gid(groupId);
+        this.artifactId = Ids.aid(artifactId);
+        this.packagingType = packagingType;
+        this.version = Ids.version(version);
+
+        this.parentCoordinate = parentCoordinate;
+
+        this.releaseDate = releaseDate;
+        this.projectName = projectName;
+
+        if (dependencies == null || dependencies.isEmpty()) {
+            this.dependencies = NO_DEPS;
+        } else {
+            // TODO check for (and prevent) "double wrapping"
```

Comment:
Could you please clarify this TODO?

Reply:
This is just a TODO for the future. I have a potential optimization in mind, but this is out-of-scope for this PR.